src/app/components/harvest/pages/list/list.component.ts

```diff
@@ -44,6 +44,55 @@ class ListComponent extends PageComponent implements OnInit {
   public ngOnInit(): void {
     this.project = this.route.snapshot.data[projectKey].model;
     this.canCreateHarvestCapability = this.project.can("createHarvest").can;
+    this.filters$ = new BehaviorSubject({
+      sorting: {
+        direction: "desc",
+        orderBy: "createdAt",
+      }
+    });
+  }
+
+  public async abortUpload(template: any, harvest: Harvest): Promise<void> {
+    const ref = this.modals.open(template);
+    const success = await ref.result.catch((_) => false);
+
+    if (success) {
+      const status: HarvestStatus = "complete";
+      this.transition(harvest, status);
+    }
+  }
+
+  /**
+   * Transition the stage of this harvest
+   *
+   * @param harvest harvest item to transition
+   * @param stage HarvestStatus to transition the harvest to
+   */
+   public transition(harvest: Harvest, stage: HarvestStatus): void {
+    this.harvestsApi
+      .transitionStatus(harvest, stage)
+      .pipe(
+        catchError((err: BawApiError) => {
+          if (err.status !== CLIENT_TIMEOUT) {
+            return throwError(() => err);
+          }
+        })
+      )
+      // We want this api request to complete regardless of lifecycle destruction
+      // eslint-disable-next-line rxjs-angular/prefer-takeuntil
+      .subscribe({
+        next: (): void => {
+          this.filters$.next({
+            sorting: {
+              direction: "desc",
+              orderBy: "createdAt"
+            }
+          });
+        },
+        error: (err: BawApiError): void => {
+          this.notifications.error(err.message);
+        },
+      });
```

Comment:
massively simplify this.

The only function you need is an `abortHarvest` function.  Copying and pasting this wholesale is a bad idea.